crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -155,16 +157,30 @@ impl Format<PyFormatContext<'_>> for FormatTrailingComments<'_> {
                                 empty_lines(lines_before_comment),
                                 format_comment(trailing)
                             ],
-                            0
+                            0 // Reserving width isn't necessary because we don't split comments and the empty lines expand any enclosing group.
                         ),
                         expand_parent()
                     ]
                 )?;
             } else {
+                // A trailing comment at the end of a line has a reserved width to consider during line measurement.
+                // ```python
+                // tup = (
+                //     "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+                // )  # Some comment
+                // ```
+
+                let normalized_comment = normalize_comment(trailing, f.context().source_code())?;
+                // SAFETY: Ruff only supports formatting files <= 4GB
+                #[allow(clippy::cast_possible_truncation)]
+                let normalized_comment_len = normalized_comment.len() as u32;
                 write!(
                     f,
                     [
-                        line_suffix(&format_args![space(), space(), format_comment(trailing)], 0),
+                        line_suffix(
+                            &format_args![space(), space(), format_comment(trailing)],
```

Comment:
We should write the comment here manually to avoid normalizing it twice. 

We probably want to factor out a helper for writing a line suffix comment:

```rust
trailing_end_of_line_comment(trailing).fmt(f)
```

Reply:
Thanks! I was thinking about this. A bounced around with a couple ideas, one of them was something like

```rust
pub(crate) const fn normalized_comment(comment_text: &str) -> FormatResult<FormatComment> {
    NormalizedComment { comment: normalized_comment(comment)? }
}
```

But `trailing_end_of_line_comment` looks like a better decision.

Reply:
First pass here https://github.com/astral-sh/ruff/compare/0aa3a86f1f641c9e67b0efa9f57924a038bb2eef..2f0f844005fa108fe627b21fab62cd1b04fd2c8e

Will do another pass on it to clean things up.